pages/api/assets/[id]/responsibilities.ts

```diff
@@ -25,20 +25,37 @@ const AssetResponsibilitiesHandler: NextApiHandler = async (req, res) => {
           id: responsibility.owner.id,
         }))
 
-        const usersRes = await Promise.all(usersWithRoles.map((user: any) => HttpClient.get(`${config.COLLIBRA_BASE_URL}/users/${user.id}`, {
+        const usersRes = await Promise.all(usersWithRoles.map((user: any) => HttpClient.get<Collibra.User>(`${config.COLLIBRA_BASE_URL}/users/${user.id}`, {
           headers: { authorization: req.headers.authorization },
         })))
 
-        const users = usersWithRoles.map((user: any) => {
-          const fullUser = usersRes.find((r) => r.body.id === user.id)?.body
+        const users = usersWithRoles.reduce((obj, user) => {
+          const collibraUser = usersRes.find((r) => r.body?.id === user.id)?.body
 
-          return {
+          if (!collibraUser) return obj
+
+          const transformedUser = {
             ...user,
-            firstName: fullUser.firstName,
-            lastName: fullUser.lastName,
-            email: fullUser.emailAddress,
+            firstName: collibraUser.firstName,
```

Comment:
A small improvement to this code is to destruct cloibraUser first and then use the result in transformedUser:

```
const { firstName, lastName, emailAddress: email } = collibraUser
const transformedUser = { ...user, firstName, lastName, email }
```

Reply:
while I agree to a certain extent, in my opinion, destructuring a lot of properties makes the code messier\